Flaky REST Gradle workflow

### Description

The REST API Gradle workflow frequently fails in GitHub Actions due to some container test setup issues.

### Steps to reproduce

1. Push a PR
2. It triggers the Gradle build workflow
3. The REST API steps periodically fail and have to be retried

### Additional context

https://github.com/hiero-ledger/hiero-mirror-node/actions/runs/14598767151/job/40951347248?pr=10991

```
ERROR: Unable to obtain connection from database (jdbc:postgresql://localhost:32786/mirror_node) for user 'mirror_node': Connection to localhost:32786 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 08001
Error Code : 0
Message    : Connection to localhost:32786 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.

Caused by: org.postgresql.util.PSQLException: Connection to localhost:32786 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
Caused by: java.net.ConnectException: Connection refused
PASS __tests__/accounts.deduplication.timestamp.test.js (7893 ms)
ERROR: Unable to obtain connection from database (jdbc:postgresql://localhost:32786/mirror_node) for user 'mirror_node': Connection to localhost:32786 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 08001
Error Code : 0
Message    : Connection to localhost:32786 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.

Caused by: org.postgresql.util.PSQLException: Connection to localhost:32786 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
Caused by: java.net.ConnectException: Connection refused
ERROR: Unable to obtain connection from database (jdbc:postgresql://localhost:32786/mirror_node) for user 'mirror_node': Connection to localhost:32786 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
SQL State  : 08001
Error Code : 0
Message    : Connection to localhost:32786 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.

Caused by: org.postgresql.util.PSQLException: Connection to localhost:32786 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
Caused by: java.net.ConnectException: Connection refused

Error: expect(received).resolves.toBeNull()

Received promise rejected instead of resolved
Rejected to value: [Error: relation "record_file" does not exist]

    at expect (node_modules/expect/build/index.js:113:15)
    at Object.expect (__tests__/service/recordFileService.test.js:70:11)
```

### Hedera network

other

### Version

main

### Operating system

None

--- a/rest/__tests__/globalSetup.js
+++ b/rest/__tests__/globalSetup.js
@@ -44,8 +44,6 @@ const initializeFlyway = () => {
   }
 };
 
-export default async function (globalConfig, projectConfig) {
-  initializeFlyway();
-}
+export default initializeFlyway;
 
 export {FLYWAY_DATA_PATH, FLYWAY_EXE_PATH, FLYWAY_VERSION};

--- a/rest/__tests__/globalSetup.js
+++ b/rest/__tests__/globalSetup.js
@@ -1,0 +1,51 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import {execSync} from 'child_process';
+import fs from 'fs';
+import path from 'path';
+
+const FLYWAY_DATA_PATH = path.join('.', 'build', 'flyway');
+const FLYWAY_EXE_PATH = path.join('.', 'node_modules', 'node-flywaydb', 'bin', 'flyway');
+const FLYWAY_VERSION = '11.8.2';
+
+const initializeFlyway = () => {
+  const flywayConfigPath = path.join(FLYWAY_DATA_PATH, `config_global.json`);
+  const flywayConfig = {
+    flywayArgs: {
+      url: 'jdbc:postgresql://127.0.0.1:-1/invalid',
+    },
+    version: FLYWAY_VERSION,
+    downloads: {
+      storageDirectory: FLYWAY_DATA_PATH,
+    },
+  };
+
+  fs.mkdirSync(FLYWAY_DATA_PATH, {recursive: true});
+  fs.writeFileSync(flywayConfigPath, JSON.stringify(flywayConfig));
+
+  let retries = 10;
+  while (retries-- > 0) {
+    try {
+      execSync(`node ${FLYWAY_EXE_PATH} -c ${flywayConfigPath} info`, {stdio: 'pipe'});
+      break;
+    } catch (e) {
+      const errMessage = e.stderr.toString();
+      if (errMessage.includes('-1 not valid')) {
+        console.log(e.stdout.toString());
+        break;
+      } else {
+        console.log(errMessage);
+      }
+    }
+  }
+
+  if (retries < 0) {
+    throw new Error('Failed to initialize flyway');
+  }
+};
+
+export default async function (globalConfig, projectConfig) {
+  initializeFlyway();
+}
+
+export {FLYWAY_DATA_PATH, FLYWAY_EXE_PATH, FLYWAY_VERSION};

--- a/rest/__tests__/integrationDbOps.js
+++ b/rest/__tests__/integrationDbOps.js
@@ -6,6 +6,7 @@ import os from 'os';
 import path from 'path';
 
 import config from '../config';
+import {FLYWAY_DATA_PATH, FLYWAY_EXE_PATH, FLYWAY_VERSION} from './globalSetup';
 import {getModuleDirname, isV2Schema} from './testutils';
 import {getPoolClass} from '../utils';
 import {PostgreSqlContainer} from '@testcontainers/postgresql';
@@ -119,8 +120,6 @@ const flywayMigrate = async (container) => {
   const containerPort = container.getPort();
   logger.info(`Using flyway CLI to construct schema for jest worker ${workerId} on port ${containerPort}`);
   const jdbcUrl = `jdbc:postgresql://${container.getHost()}:${containerPort}/${dbName}`;
-  const exePath = path.join('.', 'node_modules', 'node-flywaydb', 'bin', 'flyway');
-  const flywayDataPath = path.join('.', 'build', `${workerId}`, 'flyway');
   const flywayConfigPath = path.join(os.tmpdir(), `config_worker_${workerId}.json`); // store configs in temp dir
   const locations = getMigrationScriptLocation(schemaConfigs.locations);
 
@@ -145,13 +144,12 @@ const flywayMigrate = async (container) => {
       "url": "${jdbcUrl}",
       "user": "${ownerUser}"
     },
-    "version": "9.22.3",
+    "version": "${FLYWAY_VERSION}",
     "downloads": {
-      "storageDirectory": "${flywayDataPath}"
+      "storageDirectory": "${FLYWAY_DATA_PATH}"
     }
   }`;
 
-  fs.mkdirSync(flywayDataPath, {recursive: true});
   fs.writeFileSync(flywayConfigPath, flywayConfig);
   logger.info(`Added ${flywayConfigPath} to file system for flyway CLI`);
 
@@ -161,7 +159,7 @@ const flywayMigrate = async (container) => {
 
   while (retries-- > 0) {
     try {
-      execSync(`node ${exePath} -c ${flywayConfigPath} migrate`, {stdio: 'inherit'});
+      execSync(`node ${FLYWAY_EXE_PATH} -c ${flywayConfigPath} migrate`, {stdio: 'inherit'});
       logger.info(`Successfully executed all Flyway migrations for jest worker ${workerId}`);
       break;
     } catch (e) {

--- a/rest/jest.config.js
+++ b/rest/jest.config.js
@@ -11,6 +11,7 @@ const config = {
     '<rootDir>/node_modules/',
     '<rootDir>/__tests__/',
   ],
+  globalSetup: './__tests__/globalSetup.js',
   globalTeardown: './__tests__/globalTeardown.js',
   maxWorkers,
   reporters: [['github-actions', {silent: false}], 'jest-junit', ['summary', {summaryThreshold: 0}]],

